eva-accession-clustering/src/test/java/uk/ac/ebi/eva/accession/clustering/batch/io/ClusteringMongoReaderTest.java

```diff
@@ -0,0 +1,110 @@
+import com.mongodb.MongoClient;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.batch.item.ExecutionContext;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationContext;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+import uk.ac.ebi.eva.accession.clustering.test.configuration.MongoTestConfiguration;
+import uk.ac.ebi.eva.accession.clustering.test.rule.FixSpringMongoDbRule;
+import uk.ac.ebi.eva.accession.core.configuration.nonhuman.MongoConfiguration;
+import uk.ac.ebi.eva.accession.core.model.eva.SubmittedVariantEntity;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static junit.framework.TestCase.assertTrue;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+
+@RunWith(SpringRunner.class)
+@TestPropertySource("classpath:clustering-pipeline-test.properties")
+@UsingDataSet(locations = {"/test-data/submittedVariantEntityMongoReader.json"})
+@ContextConfiguration(classes = {MongoConfiguration.class, MongoTestConfiguration.class})
+public class ClusteringMongoReaderTest {
+
+    private static final String TEST_DB = "test-db";
+
+    private static final String ASSEMBLY = "GCA_000000001.1";
+
+    private static final int CHUNK_SIZE = 5;
+
+    private static final String SUBMITTED_VARIANT_ENTITY = "submittedVariantEntity";
+
+    private static final String CLUSTERED_SUBMITTED_VARIANT_ID = "4C1C9CE98428A4F0A6033237BA00C31E33B540D0";
+
+    private static final String NOT_CLUSTERED_SUBMITTED_VARIANT_ID = "0993FBC8C6D0A20D35B0A6DA755663B3C849676D";
+
+    private ClusteringMongoReader reader;
+
+    @Autowired
+    private MongoClient mongoClient;
+
+    @Autowired
+    private MongoTemplate mongoTemplate;
+
+    //Required by nosql-unit
+    @Autowired
+    private ApplicationContext applicationContext;
+
+    @Rule
+    public MongoDbRule mongoDbRule = new FixSpringMongoDbRule(
+            MongoDbConfigurationBuilder.mongoDb().databaseName(TEST_DB).build());
+
+    @Before
+    public void setUp() throws Exception {
+        ExecutionContext executionContext = new ExecutionContext();
+        reader = new ClusteringMongoReader(mongoClient, TEST_DB, mongoTemplate, ASSEMBLY, CHUNK_SIZE);
+        reader.open(executionContext);
+    }
+
+    @After
+    public void tearDown() {
+        reader.close();
+        mongoClient.dropDatabase(TEST_DB);
+    }
+
+    @Test
+    public void readNotClusteredSubmittedVariants() {
+        List<SubmittedVariantEntity> variants = readIntoList();
+        assertEquals(6, mongoTemplate.getCollection(SUBMITTED_VARIANT_ENTITY).count());
```

Comment:
so this is a check that the test data is ok, right? I was confused how this didn't use the variants from readIntoList,  that the readIntoList was part of the preparation of the test. No big deal but maybe it's clearer to swap these two lines?

Reply:
The line using the mongo template is reading all the variants in the test data (clustered and not clustered) and the readInto list uses the mongo reader which only takes not clustered variants. I changed the order of the lines to have better clarity